src/core/border_router/routing_manager.cpp

```diff
@@ -471,6 +482,8 @@ Error RoutingManager::PublishLocalOmrPrefix(void)
 
     OT_ASSERT(mIsRunning);
 
+    VerifyOrExit(!IsOmrPrefixAddedToLocalNetworkData());
```

Comment:
We want to skip adding the Local OMR prefix to the Local Network Data again, but should always notify server data change with `Get<NetworkData::Notifier>().HandleServerDataUpdated();`, right?

Reply:
> We want to skip adding the Local OMR prefix to the Local Network Data again, but should always notify server data change with `Get<NetworkData::Notifier>().HandleServerDataUpdated();`, right? 

No we shouldn't. It's called only when local network data is changed.

Reply:
> Will it happen that the local OMR prefix is added to local network data, but the leader network data is empty? In this case, the local OMR prefix would never be published.

In such case, the BR will detect an inconsistency between local and leader network data and automatically synchronize local network data to Leader.